### PR TITLE
Fixes #747: convert sample github_hmac to use classic Go template 

### LINKED ISSUE #747: Convert sample github_hmac to use classic Go template
Convert sample github_hmac to use classic Go template

https://github.com/openfaas/faas-cli/tree/master/sample/github_hmac

This will be less to maintain over time, a Dockerfile isn't required for the sample.

### PR DESCRIPTION
**Pull request by Amalkh5**
## Description
This change convert sample github_hmac to use classic Go template, I closed my previous PR #779  due to some issue with the branch.

## Motivation and Context

This commit slove issue #747 
- [x] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))

## How Has This Been Tested?
yes 

1. build the function
```
$ faas-cli build -f  hmac.yml
[0] > Building github-hmac.
Clearing temporary build folder: ./build/github-hmac/
Preparing ./github-hmac/ ./build/github-hmac//function
Building: alexellis2/github-hmac:latest with go template. Please wait..
Sending build context to Docker daemon  8.704kB
Step 1/30 : FROM openfaas/classic-watchdog:0.18.1 as watchdog
 ---> 94b5e0bef891
Step 2/30 : FROM golang:1.13-alpine3.11 as builder
 ---> e1fd9820be16
Step 3/30 : ARG ADDITIONAL_PACKAGE
 ---> Using cache
 ---> 9617d0823a67
Step 4/30 : ARG CGO_ENABLED=0
 ---> Using cache
 ---> bfca8715b084
Step 5/30 : ARG GO111MODULE="off"
 ---> Using cache
 ---> b19274d6603b
Step 6/30 : ARG GOPROXY=""
 ---> Using cache
 ---> 0686f28d8947
Step 7/30 : ARG GOFLAGS=""
 ---> Using cache
 ---> 06147c95ca8f
Step 8/30 : COPY --from=watchdog /fwatchdog /usr/bin/fwatchdog
 ---> Using cache
 ---> 9678e445ce73
Step 9/30 : RUN chmod +x /usr/bin/fwatchdog
 ---> Using cache
 ---> 349f42bdd8bd
Step 10/30 : ENV CGO_ENABLED=0
 ---> Using cache
 ---> 4f22b060a531
Step 11/30 : WORKDIR /go/src/handler
 ---> Using cache
 ---> f86970efa175
Step 12/30 : COPY . .
 ---> Using cache
 ---> 954c60208870
Step 13/30 : RUN cat function/GO_REPLACE.txt >> ./go.mod || exit 0
 ---> Using cache
 ---> 59041edde19e
Step 14/30 : RUN test -z "$(gofmt -l $(find . -type f -name '*.go' -not -path "./vendor/*" -not -path "./function/vendor/*"))" || { echo "Run \"gofmt -s -w\" on your Golang code"; exit 1; }
 ---> Using cache
 ---> 910813e1b53e
Step 15/30 : WORKDIR /go/src/handler/function
 ---> Using cache
 ---> 2b0708a0df5a
Step 16/30 : RUN go test ./... -cover
 ---> Using cache
 ---> d5dfd3b643a9
Step 17/30 : WORKDIR /go/src/handler
 ---> Using cache
 ---> 48fe7ead08da
Step 18/30 : RUN CGO_ENABLED=${CGO_ENABLED} GOOS=linux     go build --ldflags "-s -w" -a -installsuffix cgo -o handler .
 ---> Using cache
 ---> 35bf88a30e5e
Step 19/30 : FROM alpine:3.11
 ---> e7d92cdc71fe
Step 20/30 : RUN apk --no-cache add ca-certificates     && addgroup -S app && adduser -S -g app app     && mkdir -p /home/app     && chown app /home/app
 ---> Using cache
 ---> fb274ad8deb8
Step 21/30 : WORKDIR /home/app
 ---> Using cache
 ---> 44c32d6a9077
Step 22/30 : COPY --from=builder /usr/bin/fwatchdog         .
 ---> Using cache
 ---> 44974e0c6a3c
Step 23/30 : COPY --from=builder /go/src/handler/function/  .
 ---> Using cache
 ---> 5fcf9f954e60
Step 24/30 : COPY --from=builder /go/src/handler/handler    .
 ---> Using cache
 ---> 3380248dd0d0
Step 25/30 : RUN chown -R app /home/app
 ---> Using cache
 ---> b47f7e5c5b24
Step 26/30 : USER app
 ---> Using cache
 ---> 19b86319f55b
Step 27/30 : ENV fprocess="./handler"
 ---> Using cache
 ---> c7ce9ff36ce4
Step 28/30 : EXPOSE 8080
 ---> Using cache
 ---> 06b0cc2e825b
Step 29/30 : HEALTHCHECK --interval=3s CMD [ -e /tmp/.lock ] || exit 1
 ---> Using cache
 ---> 43b59dba4ab3
Step 30/30 : CMD ["./fwatchdog"]
 ---> Using cache
 ---> 84d3659dcba9
Successfully built 84d3659dcba9
Successfully tagged alexellis2/github-hmac:latest
Image: alexellis2/github-hmac:latest built.
[0] < Building github-hmac done in 2.35s.
[0] Worker done.

Total build time: 2.35
```
2. deplpy the function
```
$ faas-cli deploy -e secret_key=hello -e Http_X-Hub-Signature="sha1=cf23df2207d99a749a74fbe169a035e633b65d94" -f hmac.yml      
Deploying: github-hmac.
WARNING! Communication is not secure, please consider using HTTPS. Letsencrypt.org offers free SSL/TLS certificates.

Deployed. 202 Accepted.
URL: http://127.0.0.1:8080/function/github-hmac
```
3. test the function 
```
$ curl -X POST \
  http://127.0.0.1:8080/function/github-hmac \
  -d ddddd
Expected: cf23df2207d99a749a74fbe169a035e633b65d94
MessageMAC: cf23df2207d99a749a74fbe169a035e633b65d94
CalculatedMAC: b51f4760689148e80586427060c98763ed865cc4
The message was not from your GitHub account, or you don't share the same secret.
```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: sample/hmac.yml [hunk 1]
```diff
@@ -3,11 +3,12 @@ provider:
...
+  github-hmac:
+    lang: go
     fprocess: handler
-    handler: ./hmac
-    image: alexellis2/hmac
+    handler: ./github-hmac
+    image: alexellis2/github-hmac
     environment:
       secret_key: "keep-this-secret"
+      Http_X-Hub-Signature: "sha1=11"
```
Comment (alexellis) [inline sample/hmac.yml]: This should not be an environment variable in stack.yml, the idea is that it is serialised from the invocation and HTTP header.

**Comment by alexellis** [comment]
This looks better, 95% there, but one comment.

**Comment by alexellis** [comment]
We also need you to squash your 4 commits into one.

This should do it:

```
git rebase HEAD~4
```

https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History

https://thoughtbot.com/blog/git-interactive-rebase-squash-amend-rewriting-history

**Comment by Amalkh5** [comment]
@alexellis done, env var is removed and commits are squashed.

**Comment by alexellis** [comment]
Thank you for working through the feedback

**Pull request merged**